src/components/Postiterie.vue

```diff
@@ -123,7 +126,9 @@ export default {
         const croppedImagePromises = images.map((image)=>{
           return new Promise(function(resolve){
             let croppedImageData = $this.crop(image.x,image.y,image.width,image.height);
+            $this.pos.push(new Object({x: image.x, y: image.y, width: image.width, height: image.height}))
```

Comment:
Do you really need promises here? I don't see any asynchrounous calls.

Reply:
Also, you don't need `$this`. `map` inner function can return something, it'll be added to the returned array. `Promise.resolve` can take an argument which will be given when the promises resolves, in the `then` callback by instance.

You should probably dig deeper into functional javascript (map, reduce, some, every...) and Promises.

Reply:
And one more: you can probably use an autobound function then call it within the Promise handler.

Reply:
The promise is here in order to make it work with the promise.all that follows. I ran into asynch related issues (mostly because of the way trackingjs works) that the promises fixed. I'll go over these smells in details if we're ahead on the planning for the current sprint